ops/hookcmds/_secret.py

```diff
@@ -0,0 +1,242 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import datetime
+import json
+import tempfile
+from typing import (
+    Any,
+    Literal,
+    cast,
+    overload,
+)
+
+from ._types import SecretInfo, SecretRotate
+from ._utils import run
+
+
+def secret_add(
+    content: dict[str, str],
+    *,
+    label: str | None = None,
+    description: str | None = None,
+    expire: datetime.datetime | str | None = None,
+    rotate: SecretRotate | None = None,
+    owner: Literal['application', 'unit'] | None = None,
+) -> str:
+    """Add a new secret.
+
+    Args:
+        content: The content of the secret.
+        label: A label used to identify the secret in hooks.
+        description: The secret description.
+        expire: Either a duration or time when the secret should expire.
+        rotate: The secret rotation policy.
+        owner: The owner of the secret, either the application or the unit.
+    """
+    args: list[str] = []
+    if label is not None:
+        args.extend(['--label', label])
+    if description is not None:
+        args.extend(['--description', description])
+    if expire is not None:
+        if isinstance(expire, str):
+            args.extend(['--expire', expire])
+        else:
+            args.extend(['--expire', expire.isoformat()])
+    if rotate is not None:
+        args.extend(['--rotate', rotate.value])
+    if owner is not None:
+        args.extend(['--owner', owner])
+    with tempfile.TemporaryDirectory() as tmp:
+        for k, v in content.items():
+            with open(f'{tmp}/{k}', mode='w', encoding='utf-8') as f:
+                f.write(v)
+            args.append(f'{k}#file={tmp}/{k}')
+        result = run('secret-add', *args)
+    return result.strip()
+
+
+@overload
+def secret_get(
+    *,
+    id: str,
+    refresh: bool = False,
+    peek: bool = False,
+) -> dict[str, str]: ...
+@overload
+def secret_get(
+    *,
+    label: str,
+    refresh: bool = False,
+    peek: bool = False,
+) -> dict[str, str]: ...
+def secret_get(
+    *,
+    id: str | None = None,
+    label: str | None = None,
+    refresh: bool = False,
+    peek: bool = False,
+) -> dict[str, str]:
+    """Get the content of a secret.
+
+    Either the ID or the label must be provided.
+
+    Args:
+        id: The ID of the secret to retrieve.
+        label: The label of the secret to retrieve.
+        peek: Get the latest revision just this time.
+        refresh: Get the latest revision and also get this same revision for subsequent calls.
+    """
+    args: list[str] = []
+    if id is not None:
+        args.append(id)
+    if label is not None:
+        args.extend(['--label', label])
+    if refresh:
+        args.append('--refresh')
+    if peek:
+        args.append('--peek')
```

Comment:
Note from discussion with Dima, you can't use these both at the same time, the overloads should show that.

Reply:
Added.